screenshots/index.ts

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable no-console */
+import { webkit } from 'playwright'
+import fs from 'fs'
+import { build } from 'esbuild'
+
+const capitalize = (s: string): string => s[0].toUpperCase() + s.substr(1)
+
+async function go() {
+  const browser = await webkit.launch()
+  const files = fs.readdirSync('components/').map(e => e.split('.')[0])
+  // const files = ['muiDialog']
+
+  const App = `import React from 'react'
+import ReactDOM from 'react-dom'
+${files.map(e => `import ${capitalize(e)} from './components/${e}'`).join('\n')}
+import { Switch, HashRouter, Route } from 'react-router-dom'
+import Box from '@material-ui/core/Box'
+
+ReactDOM.render(<>
+  <Box p={.5}>
+  <HashRouter>
+  <Switch>
+    ${files
+      .map(e => `<Route path="/${e}" component={${capitalize(e)}} />`)
+      .join('    \n')}
+  </Switch>
+</HashRouter></Box></>,document.getElementById('root'))`
+
+  const fileName = 'app.tsx'
+  fs.writeFile(fileName, App, e => e && console.log(e))
+
+  await build({
+    entryPoints: [fileName],
+    treeShaking: true,
+    bundle: true,
+    outfile: 'app.js',
+  }).catch(() => process.exit(1))
+
+  console.log('✔︎ Compiled')
+  console.log()
+
+  for await (const file of files) {
+    console.log('✔︎ ' + file)
+    const page = await browser.newPage()
```

Comment:
it's probably faster if you declare `page` and `setViewportSize` one time outside of the loop and just call `page.goto` on it for each element.

But, it would be even better if `index.html` just renders all of the previews on one page, so you just goto `index.html` and then

```js
const previews = await page.$$('[data-snippet-preview]')
for (const preview of previews) {
  await preview.screenshot({
    path: `img/${await page.evaluate(el => el.getAttribute('data-snippet-preview'), preview)}.png`,
  })
}
```